source/extensions/common/async_files/status_after_file_error.cc

```diff
@@ -0,0 +1,64 @@
+#include "source/extensions/common/async_files/status_after_file_error.h"
+
+#include <cerrno>
+#include <string>
+
+#include "source/common/common/utility.h"
+
+#include "absl/status/status.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace AsyncFiles {
+
+absl::Status statusAfterFileError() {
+  int code = errno;
+  std::string str = Envoy::errorDetails(code);
+  switch (code) {
+  case EACCES:
+  case EPERM:
+  case EROFS:
+    return absl::PermissionDeniedError(str);
+  case EBADF:
+  case EBUSY:
+  case EISDIR:
+  case ELOOP:
+  case ENOTDIR:
+  case ETXTBSY:
+  case EWOULDBLOCK:
+    return absl::FailedPreconditionError(str);
+  case EDQUOT:
+  case EMFILE:
+  case ENFILE:
+  case ENOMEM:
+  case ENOSPC:
+    return absl::ResourceExhaustedError(str);
+  case EEXIST:
+    return absl::AlreadyExistsError(str);
+  case EFAULT:
+  case EINVAL:
+  case ENAMETOOLONG:
+    return absl::InvalidArgumentError(str);
+  case EFBIG:
+  case EOVERFLOW:
+    return absl::OutOfRangeError(str);
+  case EINTR:
+    return absl::UnavailableError(str);
+  case ENODEV:
+  case ENOENT:
+  case ENXIO:
+    return absl::NotFoundError(str);
+  case EOPNOTSUPP:
+    return absl::UnimplementedError(str);
+  case 0:
+    return absl::OkStatus();
+  default:
+    return absl::UnknownError(str);
```

Comment:
This is exactly the thing we use ENVOY_BUG for, since it would not crash in release and would crash in debug, and tell you to add a new statement to the switch. I would definitely use it here.